ts/src/xt.ts

```diff
@@ -0,0 +1,4066 @@
+         * @param {float} amount how much you want to trade in units of the base currency
+         * @param {float|undefined} price the price to fulfill the order, in units of the quote currency, can be ignored in market orders
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @param {string|undefined} params.timeInForce 'GTC', 'IOC', 'FOK' or 'GTX'
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        let timeInForce = this.safeStringUpper (params, 'timeInForce', 'GTC');
+        let response = undefined;
+        if (market['spot']) {
+            request['side'] = side.toUpperCase ();
+            request['type'] = type.toUpperCase ();
+            let marginMode = undefined;
+            [ marginMode, params ] = this.handleMarginModeAndParams ('createOrder', params);
+            const marginOrSpotRequest = (marginMode !== undefined) ? 'LEVER' : 'SPOT';
+            request['bizType'] = marginOrSpotRequest;
+            if (type === 'market') {
+                timeInForce = this.safeStringUpper (params, 'timeInForce', 'FOK');
+                if (side === 'buy') {
+                    const createMarketBuyOrderRequiresPrice = this.safeValue (this.options, 'createMarketBuyOrderRequiresPrice', true);
+                    if (createMarketBuyOrderRequiresPrice) {
+                        if (price === undefined) {
+                            throw new InvalidOrder (this.id + ' createOrder() requires a price argument for market buy orders on spot markets to calculate the total amount to spend (amount * price), alternatively set the createMarketBuyOrderRequiresPrice option to false and pass in the cost to spend into the amount parameter');
+                        } else {
+                            const amountString = this.numberToString (amount);
+                            const priceString = this.numberToString (price);
+                            const cost = this.parseNumber (Precise.stringMul (amountString, priceString));
+                            request['quoteQty'] = this.costToPrecision (symbol, cost);
+                        }
+                    } else {
+                        request['quoteQty'] = this.amountToPrecision (symbol, amount);
+                    }
+                } else {
+                    request['quantity'] = this.amountToPrecision (symbol, amount);
+                }
+            } else {
+                request['price'] = this.priceToPrecision (symbol, price);
+                request['quantity'] = this.amountToPrecision (symbol, amount);
+            }
+            request['timeInForce'] = timeInForce;
+            response = await this.privateSpotPostOrder (this.extend (request, params));
```

Comment:
since we're already splitting swap and spot order creation I think we can very easily split it into `createSpotOrder` and `createSwapOrder` and createOrder would be the bridge only. wdyt?

Reply:
Yeah I can edit createOrder in this way to split `createSpotOrder` and `createContractOrder` since delivery futures are also supported